### ISSUE TITLE
Support override resource unconditionally

### ISSUE THREAD

**Issue by n0rad** [neuspaces/terraform-provider-system#4]
When declaring a resource that already exists, the provider fail with `Error: file resource: file exists`
It could be nice to have a way to make provider's resources override existing resources unconditionally since the provider can be considered as owner of those resource.

In the same manner, when a resource handle by the provider is deleted by something else, the terraform plan is failing because the resource is not found instead of planning to re-write the resource.

**Comment by StopDenBus** [comment]
Would be nice to have this feature, maybe as  an "force-overwrite" option

**Comment by dominik-lekse** [comment]
@n0rad Thanks for reporting this issue.

I agree that there are use cases in which an existing file must be extended.

The current behavior of the `system_file` resource complies with the convention of terraform to fail when a resource already exists. In this case, terraform advises an explicit import.

I am considering an implementation of a separate resource `system_file_existant` for the use case you reported. With the separate resource, the user must explicitly decide whether an existing file is expected. The resource would automatically import an existing file and revert the file to the original state when destroyed.

**Comment by n0rad** [comment]
:wave: I agree that the default behavior should be to fail if the resource already exists, like the convention of terraform.

For resources like files and folders, I think users can face more often modifications outside of terraform than usual other terraform resource and so to me the provider should strongly support it. I don't think you have to create another resource since like @StopDenBus said, a flag on the existing resource will probably be enough, but it's up to you.

To me users setting the flag to true make it explicilt that terraform must own the content, so I don't think you have to support being able to restore the content, but again it's up to you :slightly_smiling_face:

**Comment by StopDenBus** [comment]
I agree with @n0rad. You don't need to create a new resource. Forcing overwrite content means "I know what i do and I have read the small print". 
My use-case is to install and configure a service from scratch. Mostly you will get a default configuration and you need to overwrite it's content. Manually import a resource is not an option. At the moment, I'm using resource linux_file from TelkomIndonesia/linux, but it's missing the cool md5sum feature you have.

**Comment by adminy** [comment]
did you try and import it?
